sdk/identity/Azure.Identity/src/DeviceCodeCredential.cs

```diff
@@ -62,7 +51,14 @@ public DeviceCodeCredential(string clientId, Func<DeviceCodeInfo, CancellationTo
 
             _pipeline = HttpPipelineBuilder.Build(_options);
 
-            _pubApp = PublicClientApplicationBuilder.Create(_clientId).WithHttpClientFactory(new HttpPipelineClientFactory(_pipeline)).WithRedirectUri("https://login.microsoftonline.com/common/oauth2/nativeclient").Build();
+            var pubAppBuilder = PublicClientApplicationBuilder.Create(_clientId).WithHttpClientFactory(new HttpPipelineClientFactory(_pipeline)).WithRedirectUri("https://login.microsoftonline.com/common/oauth2/nativeclient");
```

Comment:
Should the uri be in a constant?

Reply:
Yes, but there's more work than just making it a constant. It really needs to be configurable to support multiple clouds. I have created issue https://github.com/Azure/azure-sdk-for-net/issues/7932 to track this.